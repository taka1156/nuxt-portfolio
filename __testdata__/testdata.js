const MAX_NAV_DATA = 4;
const MAX_PROFILE_DATA = 4;
const MAX_CARD_DATA = 6;
const dummyImgUrl = 'http://placehold.jp/150x150.png';
const GITHUB_URL = 'https://github.com/taka1156/audio-player';
const WEBSITE_URL = 'https://takatechblog.netlify.app/';

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = () => ({
  name: 'ダミーナビゲーション',
  to: '/test1',
  img: dummyImgUrl,
});

const dummyCardTemplate = () => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
});

const dummyCardLinkTemplate = i => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  link: i % 2 === 0 ? GITHUB_URL : WEBSITE_URL,
});

const dummyProfileTemplate = () => {
  return 'ダミープロフ';
};

const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーの画像',
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, () => dummyNavTemplate());

const dummyCards = dummyFactory(MAX_CARD_DATA, () => dummyCardTemplate());

const dummyCardsLink = dummyFactory(MAX_CARD_DATA, i => dummyCardLinkTemplate(i));

const dummyProfiles = dummyFactory(MAX_PROFILE_DATA, () => dummyProfileTemplate());

const dummyLogo = 'ダミーロゴ';

export {
  dummyImg,
  dummyLogo,
  dummyRoutes,
  dummyCards,
  dummyCardsLink,
  dummyProfiles,
};
