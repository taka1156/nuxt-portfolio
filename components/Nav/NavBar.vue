<template>
  <div class="NaviBar">
    <nav v-show="!isSplashRoute">
      <div class="navbar">
        <div class="navbar__box">
          <a class="navbar__brand" href="/">Taka'sPortfolioSite</a>
          <div @click="changeState">
            <hamberger-icon :is-open="isOpen" />
          </div>
        </div>
      </div>

      <transition name="nav-fade">
        <div v-show="isOpen" class="nav__contents">
          <div class="nav__items">
            <ul class="nav__none">
              <li class="nav__item" @click="changeState">
                <nuxt-link to="/home">
                  <i class="material-icons">home</i>
                  Home
                </nuxt-link>
              </li>
              <li class="nav__item" @click="changeState">
                <nuxt-link to="/profile">
                  <i class="material-icons">account_box</i>
                  Profile
                </nuxt-link>
              </li>
              <li class="nav__item" @click="changeState">
                <nuxt-link to="/skill">
                  <i class="material-icons">edit</i>
                  Skill
                </nuxt-link>
              </li>
              <li class="nav__item" @click="changeState">
                <nuxt-link to="/portfolio">
                  <i class="material-icons">folder</i>
                  Portfolio
                </nuxt-link>
              </li>
            </ul>
          </div>
        </div>
      </transition>
    </nav>
  </div>
</template>

<script>
import HambergerIcon from "./HamburgerIcon";

export default {
  name: "NaviBar",
  components: {
    "hamberger-icon": HambergerIcon
  },
  data() {
    return {
      isOpen: false
    };
  },
  computed: {
    isSplashRoute() {
      // スプラッシュページの時はナビゲーションバー非表示
      return this.$route.path === "/";
    }
  },
  methods: {
    changeState() {
      // ナビゲーションバーの開閉管理
      this.isOpen = !this.isOpen;
    }
  }
};
</script>

<style scoped>
/* ナビゲーションバー */
.navbar {
  position: fixed;
  top: 0;
  left: 0;
  height: 60px;
  width: 100%;
  background-color: rgba(12, 12, 12, 0.9);
  z-index: 2;
}

.navbar__box {
  display: flex;
  justify-content: space-between;
  height: 90%;
  width: 100%;
  margin: 0.5em auto;
}

.navbar__brand {
  height: 30px;
  margin-top: 10px;
  font-size: 1.3em;
}

.navbar__icon {
  height: 100%;
  width: 100%;
}

.navbar__icon--open {
  animation: anime-open 0.2s linear;
}

.navbar__icon--close {
  animation: anime-close 0.2s linear;
}

/* ナビゲーションバー内容 */
.nav__contents {
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 2;
  background-color: rgba(12, 12, 12, 0.9);
}

.nav__items {
  display: flex;
  justify-content: center;
}

.nav__none {
  /* ulのスタイル無効化 */
  list-style-type: none;
  padding: 0;
}

.nav__item {
  text-align: left;
}

.nav__item,
.material-icons {
  font-size: 50px;
}

a {
  display: block;
  text-decoration: none;
  color: white;
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}

.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}

.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}

@keyframes anime-open {
  to {
    transform: rotate(0deg);
  }

  from {
    transform: rotate(-90deg);
  }
}

@keyframes anime-close {
  to {
    transform: rotate(0deg);
  }

  from {
    transform: rotate(90deg);
  }
}
</style>
