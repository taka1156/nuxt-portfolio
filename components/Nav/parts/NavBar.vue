<template>
  <div class="NavBar">
    <div class="navbar">
      <div class="navbar__box">
        <a class="navbar__brand navbar__brand--nostyle" href="/">
          Taka'sPortfolioSite
        </a>
        <div @click="changeState">
          <nav-icon :is-open="isOpen" />
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import NavIcon from './NavIcon';

export default {
  name: 'NavBar',
  components: {
    'nav-icon': NavIcon,
  },
  props: {
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
  },
  methods: {
    changeState() {
      // ナビゲーションバーの開閉管理
      this.$emit('change-state');
    },
  },
};
</script>

<style scoped>
.navbar {
  position: fixed;
  top: 0;
  left: 0;
  height: 60px;
  width: 100%;
  background-color: rgba(12, 12, 12, 0.9);
  z-index: 2;
}

.navbar__box {
  display: flex;
  justify-content: space-between;
  height: 90%;
  width: 100%;
  margin: 0.5em auto;
}

.navbar__brand {
  display: block;
  height: 30px;
  margin-top: 10px;
  font-size: 1.3em;
  color: white;
  text-decoration: none;
}
</style>
