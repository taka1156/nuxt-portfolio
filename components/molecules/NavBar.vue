<template>
  <div>
    <div class="nav-bar">
      <div class="nav-bar__box">
        <nav-logo>{{ logoText }}</nav-logo>
        <div @click="$emit('change-state')">
          <nav-icon :is-open="isOpen" />
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import NavIcon from '../atoms/NavIcon';
import NavLogo from '../atoms/NavLogo';

export default {
  name: 'NavBar',
  components: {
    'nav-icon': NavIcon,
    'nav-logo': NavLogo,
  },
  props: {
    logoText: {
      type: String,
      default: 'Please Setting Logo',
      required: true,
    },
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
  },
};
</script>

<style scoped>
.nav-bar {
  position: fixed;
  top: 0;
  left: 0;
  height: 60px;
  width: 100%;
  background-color: rgba(12, 12, 12, 0.9);
  z-index: 2;
}

.nav-bar__box {
  display: flex;
  justify-content: space-between;
  height: 90%;
  width: 100%;
  margin: 0.5em auto;
}
</style>
