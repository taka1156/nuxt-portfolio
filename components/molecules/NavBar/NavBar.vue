<template>
  <div>
    <div class="nav-bar">
      <div class="nav-bar__box">
        <base-link :route-to="`/`">{{ logoText }}</base-link>
        <base-nav-icon :is-open="isOpen" @btn-click="changeState">
          {{ isOpen ? 'CLOSE' : 'NAVI' }}
        </base-nav-icon>
      </div>
    </div>
  </div>
</template>

<script>
import BaseLink from '../../atoms/BaseLink/BaseLink';
import BaseNavIcon from '../../atoms/BaseNavIcon/BaseNavIcon';

export default {
  name: 'NavBar',
  components: {
    'base-nav-icon': BaseNavIcon,
    'base-link': BaseLink
  },
  props: {
    /**
     * ロゴテキスト
     */
    logoText: {
      type: String,
      required: true
    },
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      required: true
    }
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (開く)
       * @event changeState
       */
      this.$emit('change-state', !this.isOpen);
    }
  }
};
</script>

<style scoped>
::v-deep .base-link--extend {
  display: block;
  height: 30px;
  margin-top: 10px;
  font-size: 1.3em;
  color: white;
}

.nav-bar {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 2;
  width: 100%;
  height: 60px;
  background-color: rgba(12, 12, 12, 0.9);
}

.nav-bar__box {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 85%;
  margin: 0.5em auto;
}
</style>
