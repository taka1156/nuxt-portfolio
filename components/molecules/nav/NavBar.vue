<template>
  <div>
    <div class="nav-bar">
      <div class="nav-bar__box">
        <nav-logo>{{ logoText }}</nav-logo>
        <base-btn :use-type="`none`" @btn-click="changeState">
          <nav-icon :is-open="isOpen" />
        </base-btn>
      </div>
    </div>
  </div>
</template>

<script>
import NavLogo from '../../atoms/nav/NavLogo';
import NavIcon from '../../atoms/nav/NavIcon';
import BaseBtn from '../../atoms/base/BaseBtn.vue';
export default {
  name: 'NavBar',
  components: {
    'nav-icon': NavIcon,
    'nav-logo': NavLogo,
    'base-btn': BaseBtn,
  },
  props: {
    /**
     * ロゴテキスト
     */
    logoText: {
      type: String,
      default: 'Please Setting Logo',
      required: true,
    },
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (開く)
       * @event changeState
       */
      this.$emit('change-state', !this.isOpen);
    },
  },
};
</script>

<style scoped>
.nav-bar {
  position: fixed;
  top: 0;
  left: 0;
  height: 60px;
  width: 100%;
  background-color: rgba(40, 167, 69, 0.9);
  z-index: 2;
}
.nav-bar__box {
  display: flex;
  justify-content: space-between;
  height: 90%;
  width: 100%;
  margin: 0.5em auto;
}
</style>
