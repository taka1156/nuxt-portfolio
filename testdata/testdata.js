const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'nav-item',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png',
  },
];

const dummyProfile = [
  'TypeScriptの学習',
  'Laravel+Vueでサービスを作る',
  'AWSなどインフラ周りの学習',
  'CSS/SASSの学習',
];

const dummyCards = [
  {
    title: 'PHP',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '1~4ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Laravel',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '2~3ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'JavaScript(ES6)',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '６ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Vue.js',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '8ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
];

const dummyCardsLink = [
  {
    title: 'ブログサイト',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Nuxt.js(Vue.js), axios, microCMS, BootStrap4, vue-infinite-loading',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://takatechblog.netlify.app/',
  },
  {
    title: 'マークダウンエディター',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, BootStrap4',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/markdown-editor',
  },
  {
    title: 'LineBot',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Express.js, axios, @line/bot-sdk , Vercel(Zeit Now)',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/line-bot',
  },
  {
    title: 'オーディオプレイヤー',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, JSMediaTags',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/audio-player',
  },
];

export { dummyLogo, dummyRoutes, dummyProfile, dummyCards, dummyCardsLink };
