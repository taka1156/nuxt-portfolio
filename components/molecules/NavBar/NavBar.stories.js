import NavBar from './NavBar.vue';
import { dummyLogo } from '@/__testdata__/testdata.js';

export default {
  title: 'Molecules/NavBar',
  component: NavBar,
  argTypes: {
    isOpen: {
      control: {
        type: 'boolean'
      }
    }
  }
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { NavBar },
  template: '<nav-bar v-bind="$props" />'
});

export const Default = Template.bind({});
Default.args = { isOpen: false, logoText: dummyLogo };
