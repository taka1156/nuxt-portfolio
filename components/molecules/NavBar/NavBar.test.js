import { mount } from '@vue/test-utils';
import NavBar from './NavBar.vue';
import { dummyLogo } from '@/__testdata__/testdata.js';

describe('NavBar', () => {
  const navBar = propsData => {
    return mount(NavBar, {
      propsData: {
        ...propsData,
        logoText: dummyLogo,
      },
    });
  };

  it('NavBar初期値: isOpen, logoText', () => {
    const wrapper = navBar({ isOpen: false });
    // logoText
    expect(wrapper.vm.$options.props.logoText.required).toBe(true);
    expect(wrapper.vm.logoText).toBe(dummyLogo);
    // isOpen
    expect(wrapper.vm.$options.props.isOpen.required).toBe(true);
    expect(wrapper.vm.isOpen).toBe(false);
  });

  it('logoTextが反映されているか', () => {
    const wrapper = navBar({ isOpen: false });
    const navLogo = wrapper.find('.base-logo');
    expect(navLogo.text()).toBe(dummyLogo);
  });

  it('ナビゲーションバーを開く', () => {
    const wrapper = navBar({ isOpen: true });
    expect(wrapper.vm.isOpen).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('ナビゲーションバーを閉じる', () => {
    const wrapper = navBar({ isOpen: false });
    expect(wrapper.vm.isOpen).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('changeStateを実行した時、`change-state`のイベントが発生するか', () => {
    const wrapper = navBar({ isOpen: false });
    wrapper.vm.changeState();
    expect(wrapper.emitted('change-state')).toBeTruthy();
  });
});
