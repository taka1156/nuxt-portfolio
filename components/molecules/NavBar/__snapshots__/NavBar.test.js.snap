// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar ナビゲーションバーを閉じる 1`] = `
"<div>
  <div class=\\"nav-bar\\">
    <div class=\\"nav-bar__box\\"><a href=\\"/\\" class=\\"nav-logo\\">ダミーロゴ</a> <button class=\\"base-btn base-btn--none\\">
        <div class=\\"nav-icon\\"><span class=\\"nav-icon__top nav-icon__top--close\\"></span> <span class=\\"nav-icon__middle\\"></span> <span class=\\"nav-icon__bottom nav-icon__bottom--close\\"></span></div>
      </button></div>
  </div>
</div>"
`;

exports[`NavBar ナビゲーションバーを開く 1`] = `
"<div>
  <div class=\\"nav-bar\\">
    <div class=\\"nav-bar__box\\"><a href=\\"/\\" class=\\"nav-logo\\">ダミーロゴ</a> <button class=\\"base-btn base-btn--none\\">
        <div class=\\"nav-icon\\"><span class=\\"nav-icon__top nav-icon__top--open\\"></span> <span class=\\"nav-icon__middle nav-icon__middle--fade\\"></span> <span class=\\"nav-icon__bottom nav-icon__bottom--open\\"></span></div>
      </button></div>
  </div>
</div>"
`;
